docs: Reference Mandatory Modules Controller in Architecture Docs (#2197)
kyma-project/lifecycle-manager: docs/contributor/01-architecture.md

* [Kyma controller](../../internal/controller/kyma/controller.go) - reconciles the Kyma CR which means creating Manifest CRs for each Kyma module enabled in the Kyma CR and deleting them when modules are disabled in the Kyma CR. It is also responsible for synchronising ModuleTemplate CRs between KCP and Kyma runtimes.
* [Manifest controller](../../internal/controller/manifest/controller.go) - reconciles the Manifest CRs created by the Kyma controller, which means, installing components specified in the Manifest CR in the target SKR cluster and removing them when the Manifest CRs are flagged for deletion.

> * [Mandatory Modules controller](02-controllers.md#mandatory-modules-controllers) - reconciles the mandatory ModuleTemplates with the label `operator.kyma-project.io/mandatory-module`, selecting the highest version if duplicates exist. It translates ModuleTemplates to Manifest CRs linked to the Kyma CR, ensuring changes propagate. For removal, a deletion controller marks related Manifests, removes finalizers, and deletes the ModuleTemplate.

Reviewer: ```suggestion
* [Mandatory Modules controller](02-controllers.md#mandatory-modules-controllers) - reconciles the mandatory ModuleTemplate CRs that have the `operator.kyma-project.io/mandatory-module` label, selecting the highest version if duplicates exist. It translates the ModuleTemplate CRs to Manifest CRs linked to the Kyma CR, ensuring changes propagate. For removal, a deletion controller marks the related Manifest CRs, removes finalizers, and deletes the ModuleTemplate CR.
```
